WalletWasabi/WabiSabi/Client/HttpRequestHandlerProxy.cs

```diff
@@ -0,0 +1,74 @@
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Tor.Http;
+using WalletWasabi.Tor.Http.Extensions;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class HttpRequestHandlerProxy : IArenaRequestHandler
```

Comment:
nit: both `Http` and `Proxy` in the same class name is confusing because this does not do HTTP proxying... maybe `ArenaClientTransport` or `ArenaHttpClient` or something like that?

Reply:
Your suggestions are not convincing to me, this is why: 

* This class is a proxy (see: proxy pattern. tl;dr _"a class functioning as an interface to something else"_)
* This class communicates over http
* This class is an ArenaRquestHandler

We could replace `Proxy` by `Gateway` what is not so wrong, in that case it would be something like `HttpArenaRequestHandlerGateway`. However, in that case what would happen if we decide to drop http and use tcp? it would be `TcpArenaRequestHandlerGateway`.

I think you don't like the name because you associate the word `Proxy` with a network infrastructure component instead of a code pattern.

Reply:
I associate it with both, but I never found proxy objects to be a very coherent pattern, it means different things to different people and the implementations are very different in different packages, and i don't think in this case the OO aspects of the proxying are a bit of a reach because is just polymorphism, it doesn't wrap an `IArenaRequestHandler` it implements it, and there are multiple implementations of the same interface, this one just happens to implement that interface by dumping HTTP requests into an `IHttpClient`. OTOH, because that's such a vague term in OO, but it's a very clear term in the context of HTTP this evokes the wrong association for me. Anyway, it's not a big deal, it's just bikeshedding, we can always rename it later when you realize what a horrible and unforgivable mistake you are making ;-)